packages/tslint-config/src/rules/blueprintClassesConstantsRule.ts

```diff
@@ -29,15 +28,144 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 }
 
-function walk(ctx: Lint.WalkContext<void>): void {
-    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+function walk(ctx: Lint.WalkContext<void>) {
+    let shouldFixImports = true;
+    return ts.forEachChild(ctx.sourceFile, callback);
+
+    function callback(node: ts.Node): void {
         if (ts.isStringLiteralLike(node) || ts.isTemplateExpression(node)) {
-            const match = PATTERN.exec(node.getFullText());
-            if (match != null) {
-                // ignore first character of match: negated character class
-                ctx.addFailureAt(node.getFullStart() + match.index + 1, match[0].length - 1, Rule.FAILURE_STRING);
+            const ptMatches: string[] = [];
+            const fullText = node.getFullText();
+            let currentMatch: RegExpExecArray | null;
+            // tslint:disable-next-line:no-conditional-assignment
+            while ((currentMatch = BLUEPRINT_CLASSNAME_PATTERN.exec(fullText)) != null) {
+                const fullBlueprintName = currentMatch[1]; // e.g. pt-breadcrumb
+                const blueprintClassName = getBlueprintClassName(fullBlueprintName); // e.g. breadcrumb
+
+                // See if we should ignore this class or not.
+                if (blueprintClassName == null || shouldIgnoreBlueprintClass(blueprintClassName)) {
+                    continue;
+                } else {
+                    ptMatches.push(fullBlueprintName);
+                }
+            }
+            if (ptMatches.length > 0) {
+                // we have to fail the entire node so that we can make multiple replacements, if necessary
+                ctx.addFailureAtNode(
+                    node,
+                    Rule.FAILURE_STRING,
+                    getReplacement(node, ptMatches, ctx.sourceFile, shouldFixImports)
+                );
+                shouldFixImports = false;
+            }
+        }
+
+        return ts.forEachChild(node, callback);
+    }
+}
+
+function getReplacement(
+    node: ts.StringLiteralLike | ts.TemplateExpression,
+    ptClassStrings: string[],
+    file: ts.SourceFile,
+    shouldFixImport: boolean
+) {
+    const replacements: Lint.Replacement[] = [];
+
+    // We may need to add a blueprint import to the top of the file. We only want to do this once per file, otherwise,
+    // we'll keep stacking imports and mess things up.
+    if (shouldFixImport) {
+        replacements.push(addImportToFile(file, ["Classes"], "@blueprintjs/core"));
+    }
+
+    if (utils.isStringLiteral(node)) {
+        let stringWithoutPtClasses = node.getText();
+        ptClassStrings.forEach(cssClass => {
+            stringWithoutPtClasses = stringWithoutPtClasses.replace(cssClass, "");
+        });
```

Comment:
this would work nicely as a `reduce`. then `let stringWithoutPtClasses` could be a `const`.